File: Data Structures & Algorithms/two-integer-sum/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums: number[], target: number): number[] {
        //use a map to keep track of the needed value and its index 
        const res:number[] = []
        const map = new Map<number,number>()

        for(let i=0; i < nums.length; i++){
            const curN = nums[i]
            const need = target - curN
            if(map.has(need)){
                res[0]=map.get(need) as number
                res[1]=i;

                return res;
            }
            map.set(curN,i)
        }

        return res; 
    }
}

File: Data Structures & Algorithms/plus-one/submission-1.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        index = len(digits) - 1

        while index >= 0:
            if int(digits[index]) < 9:
                digits[index] = digits[index] + 1
                return digits
            else:
                digits[index] = 0
            index -= 1
        return [1] + digits 